### PR TITLE
Bug/dev 2947/disable foregin zipcodes

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
Disabling zip code field for foreign countries.

**Technical details:**
- Passing generateWarning to zip code field
- `ZIPField.jsx` now has `EntityWarning` component and local state to trigger showing the warning.

**JIRA Ticket:**
[DEV-2974](https://federal-spending-transparency.atlassian.net/browse/DEV-2947)

**Mockup:**
N/A

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review (if applicable)

### CODE REVIEW

File: src/js/components/search/filters/location/ZIPField.jsx [hunk 1]
```diff
@@ -43,6 +49,19 @@ export default class ZIPField extends React.Component {
...
     }
 
+    showWarning() {
+        if (!this.props.isUSA) {
+            this.setState({ showNonUsWarning: true });
+        }
+    }
+
+    hideWarning() {
+        console.log("hide warning ");
```
Comment (jonhill13) [inline src/js/components/search/filters/location/ZIPField.jsx]: @MaxwellKendall

**Comment by dlassobah** [comment]
@MaxwellKendall lets make this field consistent with other fields where input is disabled (for example, congressional district). 
• inactive fields appear to have a slight transparency effect
• inactive fields do not allow interaction or input of any kind (field and buttons disabled)

File: src/js/components/search/filters/location/ZIPField.jsx [hunk 2]
```diff
@@ -69,15 +88,24 @@ export default class ZIPField extends React.Component {
                             placeholder="Enter a ZIP code"
                             maxLength={5}
                             value={this.state.zip}
+                            onFocus={this.showWarning}
```
Comment (jonhill13) [inline src/js/components/search/filters/location/ZIPField.jsx]: Is this input component greyed out when a foreign country is selected?

File: src/js/components/search/filters/location/ZIPField.jsx (hunk 1 above)
Comment (jonhill13) [inline src/js/components/search/filters/location/ZIPField.jsx]: prolly want to remove this.

File: src/js/components/search/filters/location/ZIPField.jsx (hunk 2 above)
Comment (jonhill13) [reply src/js/components/search/filters/location/ZIPField.jsx]: although, I guess Darren's comments are pretty new.

**Comment by MaxwellKendall** [comment]
@jonhill13 responded to changes.

**Pull request merged**